internal/controlplane/handlers_authz.go

```diff
@@ -595,14 +594,34 @@ func (s *Server) removeRole(
 		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
 	}
 
+	// Validate in case there's only one admin for the project and the user is trying to remove themselves
+	if roleToRemove == authz.AuthzRoleAdmin {
+		// Get all role assignments for the project
+		as, err := s.authzClient.AssignmentsToProject(ctx, targetProject)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "error getting role assignments: %v", err)
+		}
+		// Count the number of admin roles
+		adminRolesCnt := 0
+		for _, existing := range as {
+			if existing.Role == authz.AuthzRoleAdmin.String() {
+				adminRolesCnt++
+			}
+		}
+		// If there's only one admin role, return an error
+		if adminRolesCnt <= 1 {
+			return nil, util.UserVisibleError(codes.InvalidArgument, "cannot remove the last admin from the project")
```

Comment:
**nit:** I think this should be `FailedPrecondition` rather than `InvalidArgument`.

Reply:
Thanks! Let me update it 👍